projects/react/src/lib/useMaskito.ts

```diff
@@ -32,26 +33,32 @@ export const useMaskito = ({
     options?: MaskitoOptions;
     elementPredicate?: MaskitoElementPredicate;
 } = {}): RefCallback<HTMLElement> => {
-    const [element, setElement] = useState<HTMLElement | null>(null);
+    const [predicateResult, setPredicateResult] = useState<
+        HTMLInputElement | HTMLTextAreaElement | null
+    >(null);
 
     const onRefChange: RefCallback<HTMLElement> = useCallback(
-        (node: HTMLElement | null) => {
-            setElement(node);
+        async (node: HTMLElement | null) => {
+            if (node) {
+                const predicate = await elementPredicate(node);
+
+                setPredicateResult(predicate);
+            }
         },
-        [],
+        [elementPredicate],
```

Comment:
I think it is better to make ref callback dumb as possible and without any dependencies

in your case if user changes predicate on the fly it won't automatically update